web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+namespace MOJIntranet;
+
+require_once 'prior-party-banner-events.php';
+
+defined('ABSPATH') || exit;
+
+class PriorPartyBannerEmail
+{
+
+    use PriorPartyBannerTrackEvents;
+
+    /**
+     * @var array contains all available banners
+     */
+    private array $banners = [];
+
+    /**
+     * @var string defines tha name of the ACF repeater field
+     */
+    private string $repeater_name = 'prior_political_party_banners';
+
+
+    /**
+     * @var string the name of the page for viewing banner posts
+     */
+    private string $menu_slug = 'prior-party-banners-email';
+
+    private array $post_type_labels = [];
+
+
+    public function __construct()
+    {
+        add_action('admin_menu', [$this, 'emailMenu']);
+    }
+
+    /**
+     * Loaded via a hook
+     *
+     * @return void
+     */
+
+    public function pageLoad(): void
+    {
+
+        // Get all banners from the repeater field.
+        $all_banners = get_field($this->repeater_name, 'option');
+
+        // Map the banners to a more usable format - epoch timestamps are used for comparison.
+        $mapped_banners = array_map(
+            fn ($banner) => [
+                'banner_active' => $banner['banner_active'],
+                'banner_content' => $banner['banner_content'],
+                'reference' => $banner['reference'],
+                'start_epoch' => strtotime($banner['start_date']),
+                'end_epoch' => $banner['end_date'] ? strtotime($banner['end_date']) : null
+            ],
+            $all_banners
+        );
+
```

Comment:
@EarthlingDavey not sure if he mentioned it but Damien noticed that the times were an hour behind yesterday - did you manage to sort that out?

Reply:
Oh ! I don't know, will add it to the ticket.